src/core/server/deprecations/deprecations_registry.ts

```diff
@@ -27,9 +35,21 @@ export class DeprecationsRegistry {
     dependencies: GetDeprecationsContext
   ): Promise<Array<PromiseSettledResult<DeprecationsDetails[]>>> => {
     return await Promise.allSettled(
-      this.deprecationContexts.map(
-        async (deprecationContext) => await deprecationContext.getDeprecations(dependencies)
-      )
+      this.deprecationContexts.map(async (deprecationContext) => {
+        const maybePromise = deprecationContext.getDeprecations(dependencies);
+        if (isPromise(maybePromise)) {
+          const resultOrTimeout = await withTimeout({
+            promise: maybePromise,
+            timeoutMs: this.timeout,
+          });
+          if (resultOrTimeout.timedout) {
+            throw new Error('Deprecations did not resolve in 10sec.');
+          }
+          return resultOrTimeout.value;
+        } else {
+          return maybePromise;
```

Comment:
nit: Should we treat all as promises to avoid branching code?

I can see the benefit of avoiding the timeout instrumentation if not necessary for sync operations though.

Reply:
> : Should we treat all as promises to avoid branching code? 

I wanted to, but `withTimeout` only accepts a real promise, not potential promises, so without the branching we got a TS error because `MaybePromise` does not match `Params<withTimeout>[0].promise`